### PR TITLE
chore: fix publish ci, update .gitignore

### PR DESCRIPTION
**Pull request by josxha**
- use flutter in publish CI (seems to work fine)
- fix syntax error on called workflow action (publish-single.yml)
- update .gitignore to get rid on the flutter pub publish warning
  - it seems that the .gitignore was previously copied over from the flutter/flutter repo
  - removes ~~and adds~~ one file to the repo

For more information, see the CI history here: https://github.com/josxha/flutter-maplibre-gl/actions/workflows/publish.yml

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -10,16 +10,16 @@ permissions:
 
 jobs:
   publish_maplibre_gl_platform_interface:
-    uses: ./.github/workflows/publish_single.yml
+    uses: maplibre/flutter-maplibre-gl/.github/workflows/publish-single.yml@main
```
Comment (kuhnroyal) [inline .github/workflows/publish.yml]: I think this change is not needed, only the dash in the filename was wrong :)

File: .github/workflows/publish.yml [hunk 2]
```diff
@@ -10,16 +10,16 @@ permissions:
 
 jobs:
   publish_maplibre_gl_platform_interface:
-    uses: ./.github/workflows/publish_single.yml
+    uses: .github/workflows/publish-single.yml@main
```
Comment (kuhnroyal) [inline .github/workflows/publish.yml]: And no branch. See https://stackoverflow.com/a/76614274

Comment (kuhnroyal) [inline .github/workflows/publish.yml]:
```suggestion
    uses: ./.github/workflows/publish-single.yml
```

File: .github/workflows/publish.yml (hunk 1 above)
Comment (josxha) [reply .github/workflows/publish.yml]: I tested it but the ci failed

- the @main is needed, see https://github.com/josxha/flutter-maplibre-gl/actions/runs/9181129759/workflow
- and the owner/repo seems to be required too, see https://github.com/josxha/flutter-maplibre-gl/actions/runs/9181145594

File: .github/workflows/publish.yml [hunk 3]
```diff
@@ -10,16 +10,16 @@ permissions:
...
     with:
       working-directory: maplibre_gl_platform_interface
   publish_maplibre_gl_web:
     needs: [publish_maplibre_gl_platform_interface]
-    uses: ./.github/workflows/publish_single.yml
+    uses: .github/workflows/publish-single.yml@main
```
Comment (kuhnroyal) [inline .github/workflows/publish.yml]:
```suggestion
    uses: ./.github/workflows/publish-single.yml
```

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -10,16 +10,16 @@ permissions:
...
     with:
       working-directory: maplibre_gl_web
   publish_maplibre_gl:
     needs: [publish_maplibre_gl_platform_interface, publish_maplibre_gl_web]
-    uses: ./.github/workflows/publish_single.yml
+    uses: .github/workflows/publish-single.yml@main
```
Comment (kuhnroyal) [inline .github/workflows/publish.yml]:
```suggestion
    uses: ./.github/workflows/publish-single.yml
```

File: .github/workflows/publish.yml (hunk 2 above)
Comment (josxha) [reply .github/workflows/publish.yml]: I'll try it

File: example/macos/Flutter/GeneratedPluginRegistrant.swift
Comment (kuhnroyal) [inline example/macos/Flutter/GeneratedPluginRegistrant.swift:1]: not sure about this file, don't think it should be checked in tbh

File: .github/workflows/publish.yml (hunk 2 above)
Comment (josxha) [reply .github/workflows/publish.yml]: fixed, used your solution in ceb8ba86d2067fe97628f5f25ca2f948c03fbb4a

File: .github/workflows/publish.yml (hunk 4 above)
Comment (josxha) [reply .github/workflows/publish.yml]: fixed, used your solution in ceb8ba86d2067fe97628f5f25ca2f948c03fbb4a

File: .github/workflows/publish.yml (hunk 3 above)
Comment (josxha) [reply .github/workflows/publish.yml]: fixed, used your solution in ceb8ba86d2067fe97628f5f25ca2f948c03fbb4a

File: .github/workflows/publish.yml (hunk 2 above)
Comment (josxha) [reply .github/workflows/publish.yml]: fixed, used your solution in ceb8ba86d2067fe97628f5f25ca2f948c03fbb4a

File: .github/workflows/publish.yml (hunk 1 above)
Comment (kuhnroyal) [reply .github/workflows/publish.yml]: > [Invalid workflow file: .github/workflows/publish.yml#L13](https://github.com/josxha/flutter-maplibre-gl/actions/runs/9181145594/workflow)
invalid value workflow reference: references to workflows must be prefixed with format 'owner/repository/' or './' for local workflows

File: example/macos/Flutter/GeneratedPluginRegistrant.swift
Comment (josxha) [reply example/macos/Flutter/GeneratedPluginRegistrant.swift:1]: I can remove it and git ignore it. You are right doesn't seem that necessary to include it.

Comment (kuhnroyal) [reply example/macos/Flutter/GeneratedPluginRegistrant.swift:1]: lets do it and then ill merge.

Comment (josxha) [reply example/macos/Flutter/GeneratedPluginRegistrant.swift:1]: fixed in 3b2e2a2bc9b7ed7d54181b6c9e78cce57d1cf556

**Review by kuhnroyal — APPROVED** [review]

File: .github/workflows/publish-single.yml [hunk 5]
```diff
@@ -17,13 +17,14 @@ jobs:
...
+        with:
+          cache: true
       - name: Install dependencies
-        run: dart pub get
-      # Publishing...
-      # skip validation due to https://github.com/dart-lang/setup-dart/issues/68
+        run: flutter pub get
+        # --force skips the y/N confirmation
       - name: Publish to pub.dev
-        run: dart pub publish --force --skip-validation
```
Comment (kuhnroyal) [inline .github/workflows/publish-single.yml:29]: We need to revert these changes, release is stuck here.

Comment (josxha) [reply .github/workflows/publish-single.yml:29]: - https://github.com/maplibre/flutter-maplibre-gl/pull/426
